### ISSUE TITLE
Correct way to configure the API server

### ISSUE THREAD

**Issue by caedev** [kubernetes/kubernetes-anywhere#283]
I need to enable basic auth on the API server using `--basic-auth-file [FILE]` option.   I'm guessing the best place to do this is in the phase2 `kube-apiserver.jsonnet` file, but I'm not sure where I'd put my file to ensure it's available on the node. 

Is this the right way to do it, and assuming so, how can I ensure my basic auth file is available on the master?

**Comment by colemickens** [comment]
@caedev Considering that you're going to be putting creds in that file, I think you want to handle that process somewhat out of band.

You could just stick it in one of the `phase1`s configure script, but then it persists as metadata attached to the VM that is accessible by others.

I might stick it in KeyVault and then have the configure script pull it down and place it on box.... and then edit phase2 to reference the location where you dropped it.

**Comment by caedev** [comment]
Great, thanks!

**Comment by stefanhdao** [comment]
@caedev Were you able to get this working? I added the following to the kube-apiserver.jsonnet file in phase2:

```
              "--allow-privileged",
              "--basic-auth-file=/srv/kubernetes/authfile.csv",
              "--v=4",
```

And the following to `phase1/vsphere/configure-vm.sh` in phase1:

```
echo -n "mypassword, myusername, myuserid," > "/srv/kubernetes/authfile.csv"
```

I'm getting errors saying that I need to log into the server when I try to do a basic command like "kubectl cluster-info" when transferring my config file around and trying to use the username/password combo I've chosen. I've verified that the authfile is present on the apiserver and on the master server.
